### DISCUSSION TITLE
Embedding Formatted Numbers from 'vars' Variable in Legend Text

### DISCUSSION THREAD

**Discussion by kaijk** [dbuezas/lovelace-plotly-graph-card#409] [Q&A] [answered]
I see from [here ](https://github.com/dbuezas/lovelace-plotly-graph-card?tab=readme-ov-file#extra-entity-attributes)that only `x`, `y` and `customdata:` variables can be used in `%{<variable>:.1f}` for numeric formatting.

I have a barpolar graph with no x, or y. I want to set the `name` that shows in the legend to a `vars` variable. Here is how it seems most obvious, (notwithstanding the x, y, customdata restrictions above):
```
name: "$ex 'Bearing ' + vars.windDirections.slice(-1) + '°, ' + %{vars.windSpeeds.slice(-1):.1f} + ' MPH'"
```
It doesn't work, complaining about the `%` sign being unexpected. 

This works however showing default precision:
```
name: "$ex 'Bearing ' + vars.windDirections.slice(-1) + '°, ' + vars.windSpeeds.slice(-1) + ' MPH'"
```
I've tried using customdata with no success.  It typically fails with a `missing : after property id` error.

Do I need to get `vars.windSpeeds` into `customdata`, somehow?  I tried mapping the data onto `x` 
```
x: $ex vars.windSpeeds
```
and then using `%{x:.1f}` in my `name:` assignment, but that didn't work either.

**Comment by dbuezas** [comment]
Use `vars.windSpeeds.slice(-1).toFixed(1)`

**Reply by kaijk** [reply]
This construct returns an error that, quote: `vars.windSpeeds.slice(...).toFixed is not a function`. But, it led me to find that

`Math.round(vars.windSpeeds.slice(-1) * 1e1) / 1e1`

does work for 1 decimal precision, 1e2 / 1e2 for 2 decimals, etc.

Thank you for the direction to the JS library!

**Reply by dbuezas** [reply, marked as answer]
Oh, it's probably a string then. You'd have to cast it to a number before using toFixed.
`parseFloat(vars.windSpeeds.slice(-1)).toFixed(1)`

What you did is equivalent so the best solution is the one you like the most :)
